File: frontend/src/shared/types/card-flag.ts
// Domain types for card flags — backend-agnostic

export interface CardFlagDoc {
  userId: string;
  libraryId: string;
  cardId: string;
  starred?: boolean;
  difficulty?: 'easy' | 'medium' | 'hard';
  updatedAt?: unknown;
}

export interface CardFlag extends CardFlagDoc {
  id: string;
}

export type FlagMap = Record<string, { starred?: boolean; difficulty?: 'easy' | 'medium' | 'hard' }>;

File: backend/SmartLearn.Domain/Entities/CardFlag.cs
using SmartLearn.Domain.Common;
using SmartLearn.Domain.Enums;

namespace SmartLearn.Domain.Entities;

public class CardFlag : BaseEntity
{
    public Guid UserId { get; set; }
    public Guid LibraryId { get; set; }
    public Guid CardId { get; set; }
    public bool Starred { get; set; }
    public Difficulty? Difficulty { get; set; }

    // ── Navigation ──────────────────────────────────────────
    public User User { get; set; } = null!;
    public Library Library { get; set; } = null!;
    public Card Card { get; set; } = null!;
}
